agent/ae/ae.go

```diff
@@ -93,67 +93,64 @@ func (s *StateSyncer) Run() {
 
 FullSync:
```

Comment:
This looks right, but also looks tricky to get right. Is there a unit test that exercises this part of the code already that passes before and after?